Assets/Scripts/Communication/Server.cs

```diff
@@ -0,0 +1,108 @@
+﻿using UnityEngine;
+using System;
+using System.Text;
+using System.Collections.Generic;
+using Common.Cryptography;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+
+public static class Server {
+
+	private const float SERVER_TIMEOUT = 5f;
+
+	//private static bool _timedOut;
+
+	public static void Connect() {
+		Communication.Connect(new Uri("ws://localhost:8000"));
+	}
+
+	/*public static IEnumerator Timeout(int timeout) {
+		Server._timedOut = false;
+		Debug.Log("Trying to connect to server...");
+		int seconds = 0;
+		while(seconds < timeout) {
+			seconds++;
+			yield return new WaitForSeconds(1f);
+			Debug.Log("Waited " + seconds + " of " + Server.SERVER_TIMEOUT + " seconds...");
+			if(seconds == timeout) {
+				Server._timedOut = true;
+				break;
+			}
+		}
+	}*/
+
+	public static bool Login(string username, string pw) {
+		var request = new Dictionary<string, object>();
+		request["Command"] 	= "LGN";
+		request["username"]	= username;
+		request["pw"]		= pw;
+		Communication.SendString(Json.ToString(request));
+		string response = null;
+		while(response == null) {
+			response = Communication.RecvString();
+		}
+		bool success = (bool)Json.ToDict(response)["Success"];
+		if(success) {
+			string _loginToken = Json.ToDict(response)["Token"].ToString();
+			string _encryptedToken = AES.Encrypt(_loginToken, GeneratePasswordHash());
```

Comment:
That's fine.  I'm saying to just not call Json.ToDict over and over.  Maybe create a new dictionary variable then?  You are parsing each time you call ToDict which is not necessary.